src/Promitor.Core.Scraping/Configuration/Model/AzureMetricConfiguration.cs

```diff
@@ -12,14 +16,34 @@ public class AzureMetricConfiguration
         /// </summary>
         public int? Limit { get; set; }
 
+        /// <summary>
+        ///     Information about the dimensions of an Azure Monitor metric
+        /// </summary>
+        public IReadOnlyCollection<MetricDimension> Dimensions { get; set; }
+
         /// <summary>
         ///     Information about the dimension of an Azure Monitor metric
         /// </summary>
+        [Obsolete("Dimension is deprecated, please use Dimensions instead.")]
         public MetricDimension Dimension { get; set; }
 
         /// <summary>
         ///     Configuration on how to aggregate the metric
         /// </summary>
         public MetricAggregation Aggregation { get; set; }
+
+        /// <summary>
+        ///    Checks whether the configuration contains a dimension with the given name.
+        /// </summary>
+        /// <param name="dimensionName">Dimension name to be checked for.</param>
+        /// <returns>true if the dimension name was found, false otherwise</returns>
+        public bool HasDimension(string dimensionName)
+        {
+            if (Dimension != null)
+            {
+                return Dimension.Name.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
```suggestion
                return Dimension?.Name?.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase);
```

Reply:
Added with the then also needed `?? false`

Reply:
Can you elaborate please?

Reply:
The method used to return a regular and not a nullable bool, which is why the `?? false` needed to be added in line 46 before and with this change in line 44, too. But actually, I can also just turn the return type into a nullable bool, because the places where the method is called are fine with either anyways, so I now did that to simplify it a bit.